layers/+tools/shell/funcs.el

```diff
@@ -22,11 +22,27 @@
 
 
 (defun spacemacs/projectile-shell-pop ()
-  "Open a term buffer at projectile project root."
+  "Pop-up a shell buffer at the project root.
+Customize `shell-default-shell' to control what type of shell
+buffer you create. This function will pop-up a full-width buffer
+and move your focus to it; to switch the current buffer view, use
+`spacemacs/projectile-shell'."
   (interactive)
   (let ((default-directory (projectile-project-root)))
```

Comment:
There's a pre-existing bug. When it's not in a projectile directory, this command will fail.
One solution is
```elisp
(if-let* ((backup default-directory)
          (default-directory (projectile-project-root)))
    (spacemacs/default-pop-shell)
  (message "Not in a project directory")
  (setq-local default-directory backup)
  (spacemacs/default-pop-shell))
```

Reply:
I decided instead to use `projectile-acquire-root`, which prompts the user to specify a project root when they call `spacemacs/projectile-pop-shell` outside a project directory. This matches the behavior of `projectile-run-shell` and cohorts, so will hopefully be least surprising.